lib/Query_Processor.cpp

```diff
@@ -620,123 +620,219 @@ void * get_query_digests_parallel(void *_arg) {
 	return NULL;
 }
 
-void * purge_query_digests_parallel(void *_arg) {
-	get_query_digests_parallel_args *arg = (get_query_digests_parallel_args *)_arg;
-	unsigned long long i = 0;
-	unsigned long long r = 0;
-	unsigned long long m = arg->m;
-	set_thread_name("PurgeQueryDgest", GloVars.set_thread_name);
-	for (std::unordered_map<uint64_t, void *>::iterator it=arg->gu->begin(); it!=arg->gu->end(); ++it) {
-		if ((i%DIGEST_STATS_FAST_THREADS)==m) {
-			QP_query_digest_stats *qds=(QP_query_digest_stats *)it->second;
-			delete qds;
-			r++;
+struct purge_query_digests_args {
+	umap_query_digest *digest_umap;
```

Comment:
Not a problem at the moment, but just a suggestion for future-proofing: it's a good practice to initialize pointers with `nullptr `to avoid potential issues down the line.
```suggestion
	umap_query_digest *digest_umap = nullptr;
```

Reply:
Done.